### DISCUSSION TITLE
Bonsai/BonVision parameter update questions

### DISCUSSION THREAD

**Discussion by KostasTsar** [bonsai-rx/bonsai#1178] [Q&A]
Hello everyone, kinda new to Bonsai so apologies for the probably simple questions.

Two questions I have right now are the following:
1. What is the cleanest way to make an if else statement in Bonsai? For example if I have a stimulus and I wish for it behave differently depending on a condition (e.g.  if; go up else; go down).
2. How can I use a stimulus parameter value from a frame as an input to the same stimulus in the next frame?
One way I was thinking would be similar to how you could write in a language like python:
parameter=0:
while looping through frames:
   parameter=+increment (or - depending on conditional)

The questions are for BonVision stimuli, but I would like to know how to best perform this kind of operation per frame in general.

Thank you in advance

**Comment by bruno-f-cruz** [comment]
Hey,

In a vacuum, it is kinda hard to answer any of these questions. I will leave a somewhat non-committal answer for both :P

1. For if-else, the "_bonsonic_" way to do it is to use two condition branches. The top allows values to be passed on if the incoming boolean is True, whereas the bottom one, by applying the logical operation "not" affords the opposite:
![image](https://user-images.githubusercontent.com/7049351/206687898-3bb25458-5543-4aa8-9eeb-2060206add3f.png)

Many times, however, you can shorthand a lot of conditions by using an `ExpressionTransform`. For example, if you want to return the number 5 if an incoming condition is true, and 1 otherwise you could write: `it? 5 : 0`

2. The operator you are looking for is probably `Scan` that implements a generic recursive logic. (I should note that similar to other programming languages, what you are describing is seemingly possible to be implemented in several different ways):
3. 
![image](https://user-images.githubusercontent.com/7049351/206688640-87600edd-e8cc-49b3-8b19-32dc18524ef0.png)

Hope it helps,
Bruno

**Reply by KostasTsar** [reply]
Hey Bruno,

Thank you very much, very helpful examples!

Best,
Kostas

**Reply by chiyu1203** [reply]
@KostasTsar  Hi! Did you figure out how to implement your experiment (parameter=0:
while looping through frames:
parameter=+increment (or - depending on conditional)) on Bonsai? 
I want to create a different workflow (to adjust phase differences of angle) but struggled to put "condition" into Scan. It would be wonderful if I could get some idea from you about using the Scan module.
